awx/ui_next/src/screens/Inventory/InventoryGroups/InventoryGroup/InventoryGroup.jsx

```diff
@@ -0,0 +1,134 @@
+import React, { useEffect, useState } from 'react';
+import { t } from '@lingui/macro';
+import { withI18n } from '@lingui/react';
+import { CardHeader } from '@patternfly/react-core';
+
+import { Switch, Route, withRouter, Link, Redirect } from 'react-router-dom';
+import { GroupsAPI } from '@api';
+import CardCloseButton from '@components/CardCloseButton';
+import RoutedTabs from '@components/RoutedTabs';
+import ContentError from '@components/ContentError';
+import ContentLoading from '@components/ContentLoading';
+import InventoryGroupEdit from '../InventoryGroupEdit/InventoryGroupEdit';
+import InventoryGroupDetail from '../InventoryGroupDetail/InventoryGroupDetail';
+
+function InventoryGroups({ i18n, match, setBreadcrumb, inventory, history }) {
+  const [inventoryGroup, setInventoryGroup] = useState(null);
+  const [hasContentLoading, setContentLoading] = useState(true);
+  const [hasContentError, setHasContentError] = useState(false);
+
+  useEffect(() => {
+    const loadData = async () => {
+      try {
+        const { data } = await GroupsAPI.readDetail(match.params.groupId);
+        setInventoryGroup(data);
+        setBreadcrumb(inventory, data);
+      } catch (err) {
+        setHasContentError(err);
+      } finally {
+        setContentLoading(false);
+      }
+    };
+
+    loadData();
+  }, [match.params.groupId, setBreadcrumb, inventory]);
+  const tabsArray = [
+    {
+      name: i18n._(t`Return to Groups`),
+      link: `/inventories/inventory/${inventory.id}/groups`,
+      id: 99,
+      isNestedTabs: true,
+    },
+    {
+      name: i18n._(t`Details`),
+      link: `/inventories/inventory/${inventory.id}/groups/${inventoryGroup &&
+        inventoryGroup.id}/details`,
+      id: 0,
+    },
+    {
+      name: i18n._(t`RelatedGroups`),
+      link: `/inventories/inventory/${inventory.id}/groups/${inventoryGroup &&
+        inventoryGroup.id}/nested_groups`,
+      id: 1,
+    },
+    {
+      name: i18n._(t`Hosts`),
+      link: `/inventories/inventory/${inventory.id}/groups/${inventoryGroup &&
+        inventoryGroup.id}/nested_hosts`,
+      id: 2,
+    },
+  ];
+  if (hasContentError) {
+    return <ContentError />;
+  }
+  if (hasContentLoading) {
+    return <ContentLoading />;
+  }
+  let cardHeader = hasContentLoading ? null : (
+    <CardHeader style={{ padding: 0 }}>
+      <RoutedTabs history={history} tabsArray={tabsArray} />
+      <CardCloseButton
+        linkTo={`/inventories/inventory/${inventory.id}/group`}
+      />
+    </CardHeader>
+  );
+  if (
+    !history.location.pathname.includes('groups/') ||
+    history.location.pathname.endsWith('edit')
+  ) {
+    cardHeader = null;
+  }
```

Comment:
There's two layers of conditional assignment happening for `cardHeader` and most if it is unnecessary. 

This code can't be reached if `hasContentLoading` is true, so you don't need to check it. The rest can be simplified. Try this instead:

```jsx
let cardHeader = null;
if (
  history.location.pathname.includes('groups/') &&
  !history.location.pathname.endsWith('edit')
) {
  cardHeader = (
    <CardHeader style={{ padding: 0 }}>
      <RoutedTabs history={history} tabsArray={tabsArray} />
      <CardCloseButton
        linkTo={`/inventories/inventory/${inventory.id}/group`}
     />
    </CardHeader>
  );
}
```